File: Selectionsort.py
def selection_sort(arr):
    n = len(arr)

    for i in range(n - 1):
        # Find the minimum element in the unsorted part
        min_index = i
        for j in range(i + 1, n):
            if arr[j] < arr[min_index]:
                min_index = j

        # Swap the found minimum element with the first element
        arr[i], arr[min_index] = arr[min_index], arr[i]

# Example usage:
if __name__ == "__main__":
    # Replace this list with your own list of elements
    elements = [64, 25, 12, 22, 11]

    print("Original List:", elements)

    selection_sort(elements)

    print("Sorted List:", elements)

File: Selectionsort.c
#include <stdio.h>

// Function to perform Selection Sort
void selectionSort(int arr[], int n) {
    int i, j, min_idx;
    // Traverse through the entire array
    for (i = 0; i < n-1; i++) {
        // Assume the current element as the minimum
        min_idx = i;
        // Find the minimum element in the unsorted part of the array
        for (j = i+1; j < n; j++) {
            if (arr[j] < arr[min_idx]) {
                min_idx = j;
            }
        }
        // Swap the found minimum element with the first element
        int temp = arr[min_idx];
        arr[min_idx] = arr[i];
        arr[i] = temp;
    }
}

int main() {
    // Predefined array
    int arr[] = {64, 25, 12, 22, 11};
    int n = sizeof(arr) / sizeof(arr[0]);

    // Call the selectionSort function to sort the array
    selectionSort(arr, n);

    // Print the sorted array
    printf("Sorted array: \n");
    for (int i = 0; i < n; i++) {
        printf("%d ", arr[i]);
    }
    printf("\n");

    return 0;
}
